### PR TITLE
removed extra use effects and bug

### PR DESCRIPTION
**Pull request by raymondmay95**
- Moved all of the use effects not specific to component up to its parent.
- removed use effect causing auth error
- css aligned things to better control different screens.
- partially made waveform resizable but not dynamic. It will every time the screen loads the wave form will be 70% of its div.

### CODE REVIEW

File: app/models/comment.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from .db import db
-
+from .user import User
```
Comment (raymondmay95) [inline app/models/comment.py:2]: keep me!

File: app/models/comment.py [hunk 2]
```diff
@@ -11,9 +11,10 @@ class Comment(db.Model):
     song = db.relationship("Song", back_populates="comments")
 
     def to_dict(self):
+        user = User.query.get(self.user_id)
```
Comment (raymondmay95) [inline app/models/comment.py:14]: line 14 and 17 are needed to prevent having to do another request on the frontend.

File: app/models/comment.py [hunk 3]
```diff
@@ -11,9 +11,10 @@ class Comment(db.Model):
...
         return {
             "id": self.id,
-            "userId": self.user_id,
+            "user": user.public(),
```
Comment (raymondmay95) [inline app/models/comment.py:17]: public isn't necessary but hell why not (could use to_dict)

File: app/models/user.py [hunk 4]
```diff
@@ -48,3 +48,13 @@ def to_dict(self):
             "display_name": self.display_name,
             "profile_url": self.profile_url,
         }
+
+    def public(self):
```
Comment (raymondmay95) [inline app/models/user.py:52]: wasn't sure if i needed more or less so i didn't touch the to_dict

File: react-app/src/components/songPage/addComment/addComment.module.css [hunk 5]
```diff
@@ -9,9 +9,10 @@
    box-sizing: border-box;
    top: 1rem;
    text-align: right;
-   width: 92vmax;
+   width: 86vmax;
    display: flex;
-   justify-content: flex-end;
+   justify-content: center;
```
Comment (raymondmay95) [inline react-app/src/components/songPage/addComment/addComment.module.css:14]: should help with off spacing

File: react-app/src/components/songPage/commentsList/index.js [hunk 6]
```diff
@@ -1,52 +1,13 @@
...
-        setCommenters((commenters) => [...commenters, user]);
-      } else {
-        console.log(`failed to fetch user at ${id}`);
-      }
-    }
-    userIds.forEach((id) => fetchData(id));
-  }, [userIds]);
-  if (commenters.length === list.length) {
-    // console.log("yes");
+function Index({ comments }) {
```
Comment (raymondmay95) [inline react-app/src/components/songPage/commentsList/index.js:4]: moved to parent

File: react-app/src/components/songPage/songBanner/index.js [hunk 7]
```diff
@@ -1,28 +1,14 @@
...
-    }
-    (async () => {
-      const response = await fetch(`/api/songs/${songId}`);
-      if (!response.ok) return console.log("error fetching song");
-      const fetched_song = await response.json();
-      await setSong(fetched_song);
-      return fetched_song;
-    })();
-  }, [songId]);
+function Index({ song }) {
```
Comment (raymondmay95) [inline react-app/src/components/songPage/songBanner/index.js:6]: moved to parent

File: react-app/src/components/waveform/index.js [hunk 8]
```diff
@@ -1,4 +1,10 @@
-import React, { useEffect, useState, useRef, useMemo, useCallback} from "react";
```
Comment (raymondmay95) [inline react-app/src/components/waveform/index.js:1]: dang linter

**Review by alex-pezzati — COMMENTED** [review]
looks good

**Review by alex-pezzati — APPROVED** [review]
looks good

**Pull request merged**